src/GitTabExpansion.ps1

```diff
@@ -374,8 +380,15 @@ function GitTabExpansionInternal($lastBlock, $GitStatus = $null) {
             gitAddFiles $GitStatus $matches['files']
         }
 
-        # Handles git checkout -- <path>
-        "^checkout.* -- (?<files>\S*)$" {
+        # Handles git restore -s <ref> - must come before the next regex case
+        "^restore.* -s\s*(?<ref>\S*)$" {
+            gitBranches $matches['ref'] $true
+            gitTags $matches['ref']
+            break
```

Comment:
I wonder if more of these `switch` cases should be using `break`.  Once we've found a "match" do we need to continue with every subsequent regex match?

Reply:
I was actually running into issues with this case myself and ended up matching --staged somehow, and kinda gave up and forgot to look back at it.  That said, I think this looks semi-correct. I think this will match "-S"  (which is the short-form of staged) followed by anything since we aren't using a case-insensitive match

Reply:
@rkeithhill  re: `break` ... when working on my version of this as well as #696 I realized that the fall through would end up matching some of the later cases per the regex, but then return nothing based on the inner function calls. (lots of Write-Host debugging :-p) and was going to suggest we throw in breaks. With that said, I swear I encountered at least one instance where the fall-through actually caught an edge-case.

Reply:
>  I swear I encountered at least one instance where the fall-through actually caught an edge-case.

That's why I'm nervous about throwing in `break` everywhere.  But in this new case, it should break IMO.

RE the above code matching on `-S`, good catch.  I'll make this regex be case-sensitive.